constructor/header.sh

```diff
@@ -436,7 +436,7 @@ export FORCE
 # https://github.com/ContinuumIO/anaconda-issues/issues/11148
 # First try to fix it (this apparently didn't work; QA reported the issue again)
 # https://github.com/conda/conda/pull/9073
-mkdir -p ~/.conda > /dev/null 2>&1
+test -d ~/.conda || mkdir -p ~/.conda >/dev/null 2>/dev/null || test -d ~/.conda || mkdir ~/.conda
```

Comment:
```suggestion
_maybe_mkdir_error=$(mkdir -p ~/.conda 2>&1)
if [ ! -d "~/.conda" ]; then
  echo "Could not create ~/.conda: $_maybe_mkdir_error"
  exit 1
fi
```

Let's see if this is clearer!

Reply:
```suggestion
# Avoid silent errors when $HOME is not writable
# https://github.com/conda/constructor/pull/669
test -d ~/.conda || mkdir -p ~/.conda >/dev/null 2>/dev/null || test -d ~/.conda || mkdir ~/.conda
```